cartoframes/context.py

```diff
@@ -982,36 +996,139 @@ def data_boundaries(self, df=None, table_name=None):
-        Observatory catalog
-        <https://cartodb.github.io/bigmetadata/index.html>`__ for all available
-        measures. The result of this operation is:
+    def data_discovery(self, boundary, keywords=None, regex=None, time=None):
+        """Discover data observatory meastures. This method returns the full
+        Data Observatory metadata model that is needed to create raw tables
+        or for augmenting an existing table from these measures. For the full
+        Data Observatory catalog, visit
+        <https://cartodb.github.io/bigmetadata/>.
 
-        1. It updates `table_name` by adding columns from the Data Observatory
-        2. It returns a pandas DataFrame representation of that newly augmented
-           table.
+        .. note::
+            Narrowing down a discovery query using the `keywords`, `regex`, and
+            `time` filters is important for getting a manageable metadata
+            set. Besides there being a large number of measures in the DO, a
+            metadata response can have acceptable combinations of measures with
+            demonimators (normalization and density), the same measure from
+            other years, and quantiles measurements.
+
+            For example, setting the boundary to be USA counties with no filter
+            values set will result in many thousands of measures.
+
+        Arguments:
+            boundary (str): Name of table from which the boundary is
+              calculated. Currently the extent of the table is used as the
+              boundary. Since DO measures tend to be at the country level
+              this degree of approximation is appropriate.
+            keywords (str or list of str, optional): Keywords for measures
+              to filter on. Any keyword in this list will be used to filter.
+            regex (str, optional): A regular expression to search the measure
+              descriptions. Note that this relies on PostgreSQL's case
+              insensitive operation `~*`. See `PostgreSQL docs
+              <https://www.postgresql.org/docs/9.5/static/functions-matching.html>`__
+              for more information.
 
-        Note:
-            This method alters `table_name` in the user's CARTO database by
-            adding additional columns. To avoid this, create a copy of the
-            table first and use the new copy instead.
+        Returns:
+            pandas.DataFrame: A dataframe of the complete metadata model for
+            specific measures based on the search parameters.
+        """
+        if keywords:
+            if isinstance(keywords, str):
+                keywords = [keywords, ]
+            kwsearch = ' OR '.join('numer_description ilike \'%{}%\''.format(k)
+                                   for k in keywords)
+            kwsearch = '({})'.format(kwsearch)
+
+        if regex:
+            regexsearch = '(numer_description ~* \'{}\')'.format(regex)
+
```

Comment:
Have you considered allowing a "subsection" or "topic" argument that searches the numerator tags? This way, the user can ask for all age & gender, education, or health variables, etc. You could do that two ways: by adding another filter on numer_tags OR passing it as the filter_tags argument in OBS_GetAvailableNumerators.

Reply:
It's a great idea to hook into the tag system. With this release I'm going to keep it on `numer_description` and `numer_name` only.